samples/bluetooth/direct_test_mode/src/dtm.c

```diff
@@ -1915,44 +1977,33 @@ static enum dtm_err_code on_test_transmit_cmd(uint32_t length, uint32_t freq)
 	 * packet is described in the Bluetooth Core Spsification
 	 * version 4.2 Vol. 6 Part F Section 4.1.6.
 	 */
-	nrfx_timer_compare(&dtm_inst.timer,
-			   NRF_TIMER_CC_CHANNEL0,
-			   dtm_packet_interval_calculate(
-				   dtm_inst.packet_len,
-				   dtm_inst.radio_mode),
-			   false);
+	nrfx_timer_extended_compare(&dtm_inst.timer, NRF_TIMER_CC_CHANNEL0,
+			dtm_packet_interval_calculate(dtm_inst.packet_len, dtm_inst.radio_mode),
+			NRF_TIMER_SHORT_COMPARE0_CLEAR_MASK, false);
 
 #if CONFIG_FEM
-	(void) fem_power_up();
-
 	if (dtm_inst.fem.gain != FEM_USE_DEFAULT_GAIN) {
 		if (fem_tx_gain_set(dtm_inst.fem.gain) != 0) {
 			dtm_inst.event = LE_TEST_STATUS_EVENT_ERROR;
 			return DTM_ERROR_ILLEGAL_CONFIGURATION;
 		}
 	}
 
-	(void) fem_tx_configure(nrf_timer_event_address_get(
-						dtm_inst.timer.p_reg,
-						NRF_TIMER_EVENT_COMPARE0),
-				     nrf_radio_event_address_get(
-						NRF_RADIO,
-						NRF_RADIO_EVENT_DISABLED),
-				     dtm_inst.fem.active_delay);
-#else
-	/* Configure PPI so that timer will activate radio every
-	 * 625 us
-	 */
-	nrfx_gppi_channel_endpoints_setup(
-			dtm_inst.ppi_radio_txen,
-			(uint32_t) nrf_timer_event_address_get(
-						dtm_inst.timer.p_reg,
-						NRF_TIMER_EVENT_COMPARE0),
-			nrf_radio_task_address_get(NRF_RADIO,
-						   NRF_RADIO_TASK_TXEN));
-	nrfx_gppi_channels_enable(BIT(dtm_inst.ppi_radio_txen));
+	(void)fem_power_up();
+	(void)fem_tx_configure(dtm_inst.fem.ramp_up_time);
 #endif /* CONFIG_FEM */
 
+	radio_ppi_configure(false, 0);
+
+	unsigned int key = irq_lock();
```

Comment:
Do you include `zephyr/kernel.h`? Otherwise it will trigger an error after/during current upmerge, see e.g. https://github.com/nrfconnect/sdk-nrf/pull/9193/commits/6555de3c7e8f8ee04c574ace94bcb52f36da4e23

Reply:
Ok added